02_Util/src/util/swagger.ts

```diff
@@ -117,4 +117,119 @@ export function initSwagger(
   }
 
   server.register(fastifySwaggerUi, swaggerUiOptions);
+};
+
+export const convertHeadersToMap = (
+  headers: string[],
+): Record<string, string> => {
+  const headersMap: Record<string, string> = {};
+
+  for (let i = 0; i < headers.length; i += 2) {
+    const key = headers[i];
+    const value = headers[i + 1];
+    headersMap[key] = value;
+  }
+
+  return headersMap;
+};
+
+export const getHeaderValue = (
+  headers: string[],
+  key: string,
+): string | undefined => {
+  for (let i = 0; i < headers.length; i += 2) {
+    if (headers[i].toLowerCase() === key.toLowerCase()) {
+      return headers[i + 1];
+    }
+  }
+  return undefined;
+};
+
+const getTokenFromAuthHeader = (
+  authorizationHeader: string | undefined,
+): string | undefined => {
+  if (!!authorizationHeader) {
+    const token = authorizationHeader.split('Bearer ')[1];
+    return token;
+  }
+  return undefined;
+};
+
+const getAuthorizationTokenFromRawHeaders = (
+  headers: string[],
+): string | undefined => {
+  const authorizationHeader = getHeaderValue(headers, HttpHeader.Authorization);
+  return getTokenFromAuthHeader(authorizationHeader);
+};
+
+export const getAuthorizationTokenFromRequest = (
+  request: FastifyRequest,
+): string => {
+  const token = getAuthorizationTokenFromRawHeaders(request.raw.rawHeaders);
+  if (!token) {
+    throw new UnauthorizedError('Token not found in headers');
+  }
+  return token;
+};
+
+const registerFastifyAuth = async (server: FastifyInstance) => {
+  await server.register(FastifyAuth as any).after();
+  console.log((server as any).authorization);
+
+  server.decorate(
+    'authorization',
+    function (request: any, reply: any, done: any) {
+      try {
+        const token = getAuthorizationTokenFromRequest(request);
+        if (token === '123') {
+          // todo managing real auth but for now using hard coded token
+          done(); // pass an error if the authentication fails
+        } else {
+          throw new UnauthorizedError('Token not authorized');
+        }
+      } catch (e) {
+        reply.code(HttpStatus.UNAUTHORIZED);
+        // .send(buildOcpiErrorResponse(HttpStatus.UNAUTHORIZED));
+      }
+    },
+  );
+};
+
+const registerFastifySwagger = (
+  systemConfig: SystemConfig,
+  server: FastifyInstance,
+) => {
+  server.register(fastifySwagger, {
+    openapi: {
+      info: {
+        title: 'CitrineOS Central System API',
+        description: 'Central System API for OCPP 2.0.1 messaging.',
+        version: '1.1.1',
```

Comment:
wondering if you can make this dynamic based on current version

Reply:
done 👍  and I also created this PR in OCPI to match - https://github.com/citrineos/citrineos-ocpi/pull/112